internal/client/client.go

```diff
@@ -66,6 +79,15 @@ func NewClient(cnf Config) (*Client, error) {
 	} else if usingServiceToken {
 		cnf.HttpClient.SetAuthToken(cnf.ServiceToken)
 		cnf.AuthStrategy = AuthStrategy.SERVICE_TOKEN
+	} else if usingNativeAuth {
+		token, err := Client{cnf}.OidcMachineIdentityAuth()
+
+		if err != nil {
+			return nil, fmt.Errorf("unable to authenticate with oidc machine identity [err=%s]", err)
+		}
+
+		cnf.HttpClient.SetAuthToken(token)
+		cnf.AuthStrategy = AuthStrategy.UNIVERSAL_MACHINE_IDENTITY
```

Comment:
the reuse here is intentional... in the end the schema is the same because OIDC auth is machine identity and machine identity is a single category